### ISSUE TITLE
Add reference range for metrics

### ISSUE THREAD

**Issue by mrunge** [collectd/collectd#3120]
it would be pretty good to have reference values (min/max) value reported by collectd. That would enable a quick and easy rating of "how good (or bad)" is this readout?

Since the meters should roughly know their range, I think it's best suited here, rather than higher in the stack.

Thoughts?

**Comment by faxm0dem** [comment]
you mean min/max since collectd startup?

**Comment by mrunge** [comment]
No, I meant min/max from sensor (if known). If your temperature sensor can report values between
+20 and +90 degrees celsius, then it should report that. That would help you to know if your cpu is rather hot or not.

Same for networks; the network card *knows* how many packages can be delivered through the interface.

**Comment by sunkuranganath** [comment]
@mrunge good suggestion. This could be an easy and simple way of validating a metric.
One way to look at this is to get the valid range based on the data type of that metric. For example, a counter with "int" data type has a bounds of "Integer". As long as the value is with in that range, its a valid value. Some metrics are of data type hex or character. Its difficult to provide exact reference range for those.
However if the metrics are to to fit into a specification, example temperature range of +20 to +90 degrees, its simpler to report the range.

How do you suggest these be reported out? In metadata or a custom field?

**Comment by mrunge** [comment]
The idea here is more to enable a much easier "rating" on metric readout. "how good is this value?"
Since you could also do this at a central location, like in influx, graphite or prometheus, you have to know the reference values for a given metric data.

Regarding metadata or custom field: I don't really care; I would think it's a unnecessary duplication of data, if the reference min/max would be exported for each readout.

**Comment by rpv-tomsk** [comment]
Let's look into mysql as example.
MySQL server has a metric such as "threads connected". For this metric, 0 is min value and `max_connections` variable value is max value for metric. 

`max_connections` can be changed at any time while MySQL server is running, invoking command `set global max_connections=XXX`. So, Collectd has to query it's value 'often enough'.

Typical case is one Collectd is "agent" and another Collectd is a hub.
So, we need to update "max value" on "hub" side.
If "agent" will send this value only on value change, then "hub" has to store this value into some storage to get in after "hub" restart (while "agent" will not send this value unless it changed).

Or, we has to use persistent connections (TCP, not unidirectional UDP) between HUB and AGENT, so AGENT could detect HUB restart and resend min/max values.

**Comment by mrunge** [comment]
Ugh, good catch! That'll not be as easy as I thought then.

Let's take the mysql example, 0 as min value for max_connections would be fine, and IMHO it should also be fine to keep max undefined (or -1) to express: there is no explicit maximum, or to state: I can't or don't know.
